sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Internal utility functions for extendable enumerations.
+ *
+ */
+#include <string>
+
+namespace Azure { namespace Core { namespace _internal {
+  /** @brief Template base class helper for implementing extendable enumerations.
+   *
+   * This template exists to simplify the experience of authoring ["extendable
+   * enumerations"](https://azure.github.io/azure-sdk/cpp_implementation.html#cpp-enums).
+   *
+   * An extendable enumeration derives from the #ExtendableEnumeration base class passing in the
+   * extendable enumeration type as the template specialization.
```

Comment:
```suggestion
   * extendable enumeration type as the template argument.
```
Time to be pedantic! In:
```c++
template <class T>
class C {};
C<int> c;
```
`C` is a class template, `T` is a template type parameter, `C<int>` is a specialization of `C`, and `int` is a template type argument.

Reply:
Should we say "derives publicly from" for clarity?